### ISSUE TITLE
depModuleVersion 204 is missing

### ISSUE THREAD

**Issue by jlkalberer** [particle-iot/device-os#1942]
<details>
  <summary><i>The firmware release on Github seems to be an incorrect build. It seems like the module metadata is incorrect. New firmware like 1.4.0 has a `depModuleVersion` of 204 which doesn't exist anywhere in the releases.</i></summary>

</details>

### Expected Behavior

There should be a 204 release in order to satisfy dependencies or the `depModuleVersion` should point at 207.

### Observed Behavior

[When I use `HalModuleParser` on 1.4.1 bootloader this is what I get](https://github.com/particle-iot/binary-version-reader)
```
{
    "filename": "p1-system-part1@1.4.1-rc.1.bin",
    "crc": {
      "ok": 1,
      "storedCrc": "f2693482",
      "actualCrc": "f2693482"
    },
    "prefixInfo": {
      "moduleStartAddy": "8020000",
      "moduleEndAddy": "805a024",
      "moduleFlags": 0,
      "moduleVersion": 1402,
      "platformID": 8,
      "moduleFunction": 4,
      "moduleIndex": 1,
      "depModuleFunction": 4,
      "depModuleIndex": 2,
      "depModuleVersion": 204,
      "dep2ModuleFunction": 0,
      "dep2ModuleIndex": 0,
      "dep2ModuleVersion": 0
    },
    "suffixInfo": {
      "productId": 65535,
      "productVersion": 65535,
      "fwUniqueId": "6af29ee148779e0de5627653c21aec4633bf3e512bfc3cc62b991643d7149648",
      "reserved": 0,
      "suffixSize": 40,
      "crcBlock": "f2693482"
    }
  },
```

### Steps to Reproduce

[I can reproduce this in spark-protocol](https://github.com/Brewskey/spark-protocol) by running `yarn update-firmware`. The `settings.json` file has all the parsed modules. 

[The bug was reported here](https://github.com/Brewskey/spark-protocol/issues/145)
I've added a hack to update `depModuleVersion` to 207 but it doesn't seem safe.

### Test App

```js
const parser = new HalModuleParser();
parser.parseFile(
  'bootloader-0.7.0-p1.bin',
  (result:) => {
    console.log(result);
  },
 )
```

**Comment by rickkas7** [comment]
The bootloader version is not the same as the device OS system binary version.

https://github.com/particle-iot/device-os/blob/develop/system/system-versions.md

The bootloader is version 101 for 0.7.0. 
The system-part1 and system-part2 should be version 207 for the final release of 0.7.0.

**Comment by jlkalberer** [comment]
@rickkas7 - I've updated my comment a bit.

The issue is that 1.4.0 and newish firmware have a `depModuleVersion` of 204. This module build doesn't exist anywhere in the releases AFAIK.

**Comment by avtolstoy** [comment]
@jlkalberer Thanks for reporting the issue. It seems that we did not publish any binaries for 0.7.0-rc.5 (which has the `204` system parts): https://github.com/particle-iot/device-os/releases/tag/v0.7.0-rc.5

Just so that we are on the same page:
```
      "depModuleFunction": 4, // MODULE_FUNCTION_SYSTEM_PART
      "depModuleIndex": 2, // system-part2
      "depModuleVersion": 204, // version 204 (0.7.0-rc.5)
```

This release does exist in the database, so when performing cloud OTA everything should work out as intended, however the main purpose why `system-part1` has a strict requirement on `system-part2` of a particular version is to prevent the device from directly upgrading from < 0.7.0 releases to prevent the device getting bricked. I agree though, that we should bump it to 207 (0.7.0 final) and we'll do that before 1.4.1 final.

For your particular case:

> I've added a hack to update depModuleVersion to 207 but it doesn't seem safe.

It is safe to do so or even simply having 0.7.0 final (207) binaries in place of 0.7.0-rc.5 (204).

**Comment by jlkalberer** [comment]
Thanks, for clearing that up.

Any way you can publish those binaries?
I maintain the latest version of the Particle local cloud and I'd like to revert my hack in case it introduces some bugs.

**Comment by avtolstoy** [comment]
> Any way you can publish those binaries?
> I maintain the latest version of the Particle local cloud and I'd like to revert my hack in case it introduces some bugs.

I just double-checked and we've pulled that release (0.7.0-rc.5) due a critical issue with it. You should probably be using 0.7.0 final in its place.

We've also fixed the module dependency in 1.4.1: https://github.com/particle-iot/device-os/blob/develop/modules/shared/system_module_version.mk#L9

I'm going to close this issue, please feel free to reopen it if any additional problems arise.
